src/xai_grok/providers/unauthorized_user_error_provider.py

```diff
@@ -0,0 +1,18 @@
+import http
+
+import xai_grok.errors as errors
+
+
+class UnauthorizedUserErrorProvider:
+
+    _NO_API_KEY_PROVIDED_MESSAGE = "Invalid Authorization header 'Bearer'"
+    _INVALID_API_KEY_PROVIDED_MESSAGE = "Incorrect API key provided"
+
+    def provide_error(self, response: dict) -> errors.UnauthorizedUserError:
+        if response.status_code == http.client.BAD_REQUEST:
```

Comment:
The `response` parameter is typed as `dict` but is used as an object with `status_code` attribute and `json()` method, indicating it should be a response object instead.